src/routers/upload.py

```diff
@@ -297,25 +295,76 @@ async def upload_geotiff_chunk(
 				initial_dataset,
 			)
 		)
-		# loop = asyncio.get_running_loop()
-		# loop.run_in_executor(
-		# 	executor,
-		# 	combine_chunks,
-		# 	tmp_dir,
-		# 	chunks_total,
-		# 	file_name,
-		# 	target_path,
-		# 	token,
-		# 	initial_dataset,
-		# )
-
-		# executor.submit(combine_chunks, tmp_dir, chunks_total, file_name, target_path, token, initial_dataset)
 
 		return initial_dataset
 
 	return {'message': f'Chunk {chunk_index} of {chunks_total} received'}
 
 
+@router.put('/datasets/{dataset_id}/label-object')
+async def upload_label_object(
+	file: UploadFile,
+	dataset_id: int,
+	user_id: Annotated[str, Form()],
+	file_type: Annotated[str, Form()],
+	file_alias: Annotated[str, Form()],
+	label_description: Annotated[str, Form()],
+	token: Annotated[str, Depends(oauth2_scheme)],
+):
+	"""
+	Upload a label object.
+	"""
+
+	user = verify_token(token)
+	if not user:
+		raise HTTPException(status_code=401, detail='Invalid token')
+	logger.info(f'Received label object for dataset {dataset_id} from user {user_id}', extra={'token': token})
+
+	# create folder if not exists settings.labels_objects_path / dataset_id
+	if not (settings.label_objects_path / str(dataset_id)).exists():
+		(settings.label_objects_path / str(dataset_id)).mkdir(parents=True, exist_ok=True)
+	# count number of files in the folder
+	num_files = len(list((Path(settings.label_objects_path) / str(dataset_id)).glob('*'))) + 1
+
+	target_path = settings.label_objects_path / str(dataset_id) / f'{file_alias}_{num_files}.{file_type}'
```

Comment:
I would prefer adding an upload date, instead of just counting the objects here. Serves the same purpose but is somewhat more helpful. like `datetime.datetime.now().isoformat()` or something

Reply:
I just realized that its impossible to upload multiple files to a specific dataset, since the dataset_id will always increment, so the counting is not necessary. The same is true for the additional folder, but I would keep it this way, since its cleaner.

Reply:
Even better.

Then you can remove counting Code

Reply:
I will keep the date, in case we or later the application want to upload additinal files later.